src/source-dpdk.c

```diff
@@ -0,0 +1,640 @@
+#else
+    SCLogWarning(SC_ERR_TM_THREADS_ERROR, "NUMA node retrieval is not supported on this OS.");
+#endif
+
+    return node;
+}
+
+/**
+ * \brief Registration Function for ReceiveDPDK.
+ * \todo Unit tests are needed for this module.
+ */
+void TmModuleReceiveDPDKRegister(void)
+{
+    tmm_modules[TMM_RECEIVEDPDK].name = "ReceiveDPDK";
+    tmm_modules[TMM_RECEIVEDPDK].ThreadInit = ReceiveDPDKThreadInit;
+    tmm_modules[TMM_RECEIVEDPDK].Func = NULL;
+    tmm_modules[TMM_RECEIVEDPDK].PktAcqLoop = ReceiveDPDKLoop;
+    tmm_modules[TMM_RECEIVEDPDK].PktAcqBreakLoop = NULL;
+    tmm_modules[TMM_RECEIVEDPDK].ThreadExitPrintStats = ReceiveDPDKThreadExitStats;
+    tmm_modules[TMM_RECEIVEDPDK].ThreadDeinit = ReceiveDPDKThreadDeinit;
+    tmm_modules[TMM_RECEIVEDPDK].cap_flags = SC_CAP_NET_RAW;
+    tmm_modules[TMM_RECEIVEDPDK].flags = TM_FLAG_RECEIVE_TM;
+}
+
+/**
+ * \brief Registration Function for DecodeDPDK.
+ * \todo Unit tests are needed for this module.
+ */
+void TmModuleDecodeDPDKRegister(void)
+{
+    tmm_modules[TMM_DECODEDPDK].name = "DecodeDPDK";
+    tmm_modules[TMM_DECODEDPDK].ThreadInit = DecodeDPDKThreadInit;
+    tmm_modules[TMM_DECODEDPDK].Func = DecodeDPDK;
+    tmm_modules[TMM_DECODEDPDK].ThreadExitPrintStats = NULL;
+    tmm_modules[TMM_DECODEDPDK].ThreadDeinit = DecodeDPDKThreadDeinit;
+    tmm_modules[TMM_DECODEDPDK].cap_flags = 0;
+    tmm_modules[TMM_DECODEDPDK].flags = TM_FLAG_DECODE_TM;
+}
+
+static inline void DPDKDumpCounters(DPDKThreadVars *ptv)
+{
+    struct rte_eth_stats eth_stats;
+    int retval = rte_eth_stats_get(ptv->port_id, &eth_stats);
+    if (unlikely(retval != 0)) {
+        SCLogError(SC_ERR_STAT, "Failed to get stats for port id %d: %s", ptv->port_id,
+                strerror(-retval));
+        return;
+    }
+
+    uint64_t th_pkts = StatsGetLocalCounterValue(ptv->tv, ptv->capture_dpdk_packets);
+    StatsAddUI64(ptv->tv, ptv->capture_dpdk_packets, ptv->pkts - th_pkts);
+    SC_ATOMIC_ADD(ptv->livedev->pkts, ptv->pkts - th_pkts);
+
+    /* Some NICs (e.g. Intel) do not support queue statistics and the drops can be fetched only on
+     * the port level. Therefore setting it to the first worker to have at least continuous update
+     * on the dropped packets. */
+    if (ptv->queue_id == 0) {
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_rx_errs,
+                eth_stats.imissed + eth_stats.ierrors + eth_stats.rx_nombuf + ptv->pkts);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_imissed, eth_stats.imissed);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_rx_no_mbufs, eth_stats.rx_nombuf);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_ierrors, eth_stats.ierrors);
+        StatsSetUI64(ptv->tv, ptv->capture_dpdk_tx_errs, eth_stats.oerrors);
+        SC_ATOMIC_SET(ptv->livedev->drop, eth_stats.imissed + eth_stats.ierrors +
+                                                  eth_stats.rx_nombuf + eth_stats.oerrors +
+                                                  ptv->pkts);
```

Comment:
Hey! Seems like we're adding `ptv->pkts` to the `ptv->livedev->drop` and the `ptv->capture_dpdk_rx_errs` mistakenly?
AFAICT `ptv->pkts` is just the count of packets received from `rte_eth_rx_burst` (in `ReceiveDPDKLoop`), and shouldn't be included in error/drop counts?